### ISSUE TITLE
[alpha2] dateAlignment: "year" fails in Safari

### ISSUE THREAD

**Issue by christiankn** [fullcalendar/fullcalendar#4363]
Using DateAlignment: "year" in the configuration fails in Safari (12). 

The cause is that it will attempt to initialize a DateMarker with an array containing only the year. This works in most browsers, but gives invalid date in Safari.

`new Date(Date.UTC.apply(Date, [2018]))`

The issue is in datelib/env.ts:323 
```
startOfYear(m: DateMarker): DateMarker {
    return this.calendarSystem.arrayToMarker([
      this.calendarSystem.getMarkerYear(m)
    ])
  }
```
Appending `0` to the array would fix this issue.

_Minimum example:_
https://codepen.io/anon/pen/wYbJEM?&editors=001
Try in Safari and a different browser to see.

**Comment by acerix** [comment]
Thanks for the report. I don't have Safari to test, but I did find that the `monthIndex` is required when creating a date so it should be specified.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date#Syntax

**Comment by arshaw** [comment]
thanks for the on-point diagnosis of this bug.

i've fixed this in the `v4` branch: https://github.com/fullcalendar/fullcalendar/commit/90eb3f66db3491bc2c96cbd824b1f38f8616bcff

I'll comment on this ticket again when I release the fix. Closing

**Comment by arshaw** [comment]
this has been addressed and released in [4.0.0-alpha.4](https://github.com/fullcalendar/fullcalendar/releases/tag/v4.0.0-alpha.4)
